packages/ui/src/components/numpad/numpad.native.tsx

```diff
@@ -0,0 +1,159 @@
+import { ReactNode } from 'react';
+
+import { ArrowLeftIcon } from '../../icons/arrow-left-icon.native';
+import { Box } from '../box/box.native';
+import { NumpadKey, NumpadKeySlot } from './numpad-key.native';
+
+const FALLBACK_LOCALE_IDENTIFIER = 'en';
+
+type Mode = 'numeric' | 'decimal';
+type DecimalSeparator = '.' | ',';
+type KeyId =
+  | '1'
+  | '2'
+  | '3'
+  | '4'
+  | '5'
+  | '6'
+  | '7'
+  | '8'
+  | '9'
+  | '0'
+  | 'delete'
+  | 'decimalSeparator';
+type Layout = [KeyId | null, KeyId | null, KeyId | null][];
+
+interface Key {
+  id: KeyId;
+  accessibilityLabel: string;
+  element: ReactNode;
+}
+
+function getAllKeys(locale: string): Record<KeyId, Key> {
+  const decimalSeparator = getDecimalSeparator(locale);
+
+  return {
+    '1': { id: '1', accessibilityLabel: '1', element: '1' },
+    '2': { id: '2', accessibilityLabel: '2', element: '2' },
+    '3': { id: '3', accessibilityLabel: '3', element: '3' },
+    '4': { id: '4', accessibilityLabel: '4', element: '4' },
+    '5': { id: '5', accessibilityLabel: '5', element: '5' },
+    '6': { id: '6', accessibilityLabel: '6', element: '6' },
+    '7': { id: '7', accessibilityLabel: '7', element: '7' },
+    '8': { id: '8', accessibilityLabel: '8', element: '8' },
+    '9': { id: '9', accessibilityLabel: '9', element: '9' },
+    '0': { id: '0', accessibilityLabel: '0', element: '0' },
+    decimalSeparator: {
+      id: 'decimalSeparator',
+      accessibilityLabel: decimalSeparator,
+      element: decimalSeparator,
+    },
+    delete: { id: 'delete', accessibilityLabel: 'delete', element: <ArrowLeftIcon /> },
+  };
+}
+
+const layouts: Record<Mode, Layout> = {
+  numeric: [
+    ['1', '2', '3'],
+    ['4', '5', '6'],
+    ['7', '8', '9'],
+    [null, '0', 'delete'],
+  ],
+  decimal: [
+    ['1', '2', '3'],
+    ['4', '5', '6'],
+    ['7', '8', '9'],
+    ['decimalSeparator', '0', 'delete'],
+  ],
+};
+
+function getLayout(mode: Mode, locale: string): (Key | null)[] {
+  return layouts[mode].flat().map(id => {
+    if (!id) {
+      return null;
+    }
+
+    return getAllKeys(locale)[id];
+  });
+}
+
+function getDecimalSeparator(locale: string): DecimalSeparator {
+  try {
+    const formatter = new Intl.NumberFormat(locale);
```

Comment:
An open question here, but do we want to allow for different formats? Will we have to normalize them once we start signing transactions and does that add complexity for us? @fbwoolf/@kyranjamie curious on your thoughts.

Reply:
Great question. The numpad, and the display component that will feed off of it, deals with the string representation of the number in the correct format for based on the user's locale: period for English, comma for most non-English speaking users. This is the visual aspect of the interaction, it really only has to be a string because of intermediate trailing decimal zeros, i.e. the user is typing "12.00...123". Transforming this to a number further down the line regardless of the format is likely going to be trivial.